### DISCUSSION TITLE
enh: Enhance websearch

### DISCUSSION THREAD

**Discussion by manfromdownunder** [open-webui/open-webui#3426] [General]
**Is your feature request related to a problem? Please describe.**
Websearch is literal, meaning if I am talking about cats, and I websearch how fast can they run, then the search is "how fast can they run" and returns results not relating to the cats or context of the conversation.

**Describe the solution you'd like**
Websearch should summarise chat history then use NLP to submit a search query based on the summary and last question asked for actual relevant/useful results.

**Describe alternatives you've considered**
I have not considered any alternatives.

**Additional context**
Not at this stage. I just think that we should have an NLP layer between the question and the search query.

**Comment by morbificagent** [comment]
This is exactly my "problem" at the moment... Was speaking about the hybrid option here in openwebui and because the search didnt know anything about the conversation it gave back hybrid-car information... 

Is there eventually a tool/function/pipe at the moment for that?
